src/bindings/python/src/pyopenvino/utils/utils.cpp

```diff
@@ -404,6 +404,20 @@ ov::AnyMap py_object_to_any_map(const py::object& py_obj) {
     return return_value;
 }
 
+template <typename... Args, std::size_t... I>
+std::tuple<Args...> tuple_from_py_tuple_impl(const py::tuple& py_tuple, std::index_sequence<I...>) {
+    return std::make_tuple(py_tuple[I].cast<Args>()...);
+}
+
+template <typename... Args>
+std::tuple<Args...> tuple_from_py_tuple(const py::tuple& py_tuple) {
+    if (py_tuple.size() != sizeof...(Args)) {
+        OPENVINO_THROW(false, "Size of py::tuple does not match size of std::tuple");
+    }
+
+    return tuple_from_py_tuple_impl<Args...>(py_tuple, std::index_sequence_for<Args...>{});
+}
+
```

Comment:
try to simplify it a little bit
for example:

```suggestion
template <typename... Args>
std::tuple<Args...> tuple_from_py_tuple(const py::tuple& py_tuple) {
    if (py_tuple.size() != sizeof...(Args)) {
        OPENVINO_THROW(false, "Size of py::tuple does not match size of std::tuple");
    }
    return std::make_tuple(py_tuple[py_tuple.size()].cast<Args>()...)
}

```

Reply:
`tuple_from_py_tuple_impl` moved inside `tuple_from_py_tuple` as lambda

Reply:
template lambda is not part of C++17.

Reply:
I reverted that lambda. Some CI's were failing. I believe `std::index_sequence<I...>` has to be passed as template argument here to deduce indexes